File: Design_Questions/Snake_And_Ladder/Player.java
package Design_Questions.Snake_And_Ladder;
public class Player {
    private final String name;
    private int currentPosition;
    public Player(String name){
        this.name=name;
        currentPosition=0;
    }
    public String getName(){
        return name;
    }
    public int getPostion(){
        return currentPosition;
    }
    public void setPosition(int newPosition){
        this.currentPosition=newPosition;
    }
}

File: Design_Questions/Snake_And_Ladder_CPP/Player.cpp
#include "Player.hpp"
#include<string>
using namespace std;
Player::Player(const string& name):playerName(name),currentPosition(0){}
string Player::getName() const{
    return playerName;
}
int Player::getPosition() const{
    return currentPosition;
}
void Player::setPosition(int newPosition){
    currentPosition = newPosition;
}
